### PR TITLE
Lets the bureaucracy mishap event have a chance to roll rare jobs

### PR DESCRIPTION
**Pull request by Cocacolagua**
## What Does This PR Do
Detective is removed from the blacklist for the event, and can roll normally.

The following jobs:
- Captain
- Head of Personnel
- Head of Security
- Research Director
- Chief Engineer
- Chief Medical Officer
- Nanotrasen Representative
- Magistrate
- (you became a head after this pr was made so you get a seat here I guess?) Quartermaster

have been removed from the blacklist and added to the "rare" category of jobs, which can roll for the event if they get selected normally AND pass a 40% chance, this category can only have their job slots increased, not decreased.

## Why It's Good For The Game
The event is pretty sad, and misses the point on why it works on other servers. I understand that it is a minor event, but it rarely has any impact on the round at all. This change aims to let the jobs that can cause the most impact be considered for the event at a rarer chance.

If you're wondering what the chances of it rolling a rare job are:
After this change, there will be 25 common jobs and 8 rare jobs (33 jobs total), therefore each roll will have a 24.24% of selecting any rare job, but then you have to pass the additional 40% chance, which puts us at around 9.6% of actually rolling the thing.
The event has a 50/50 between it affecting 1 or 2 jobs, so on average a 14.4% of rolling a rare job. 
Additionally, when a job is picked to be affected, it chooses between (-2,-1,1,2) on what to do with the job slots, and given that this change does not deduct said slots, you will have a 75% of it only increasing one slot, and a 25% of it increasing two (due to how the formula is done).

## Testing
1. Booted up game
2. Opened Event Manager and started SPAMMING that shit
3. Waited a bit for it to roll a rare job

## Changelog
:cl:
tweak: Detective can now roll for the bureaucratic error event.
tweak: Captain, Heads of Staff, Magistrate, and Nanotrasen Representative have a low chance to roll for the bureaucratic error event, they can only have their slots increased.
/:cl:

### CODE REVIEW

File: code/modules/events/bureaucratic_error.dm [hunk 1]
```diff
@@ -3,21 +3,24 @@
 	/// Jobs that are not allowed to be picked for the bureaucratic error
 	var/list/blacklisted_jobs = list(
 		/datum/job/assistant,
+		/datum/job/ai,
+		/datum/job/cyborg,
+		/datum/job/blueshield,
+		/datum/job/chaplain,
+		/datum/job/officer,
+		/datum/job/warden,
```
Comment (Burzah) [inline code/modules/events/bureaucratic_error.dm:11]:
```suggestion
		/datum/job/warden
```

File: code/modules/events/bureaucratic_error.dm [hunk 2]
```diff
@@ -3,21 +3,24 @@
...
 		/datum/job/cmo,
 		/datum/job/rd,
 		/datum/job/hos,
-		/datum/job/ai,
-		/datum/job/cyborg,
 		/datum/job/captain,
 		/datum/job/hop,
 		/datum/job/nanotrasenrep,
-		/datum/job/blueshield,
 		/datum/job/judge,
```
Comment (Burzah) [inline code/modules/events/bureaucratic_error.dm:23]:
```suggestion
		/datum/job/judge
```

**Review by Burzah — COMMENTED** [review]
Forgive the little bit of nitpicking here.

**Review by DGamerL — COMMENTED** [review]
Just do Burza's review and it should be good

**Review by Burzah — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**